packages/dev/core/src/Rendering/GaussianSplatting/gaussianSplatting.ts

```diff
@@ -284,36 +310,47 @@ export class GaussianSplatting {
             this.dispose();
         }
         this._setData(new Uint8Array(data as any));
-        const matricesData = new Float32Array(this.vertexCount * 16);
+        const matricesData = new Float32Array(this.vertexCount * 1); // matrix is only used to allocate instances
+        const splatIndex = new Float32Array(this.vertexCount * 1);
 
         const updateInstances = (indexMix: Uint32Array) => {
             for (let j = 0; j < this.vertexCount; j++) {
-                const i = indexMix[2 * j];
-                const index = j * 16;
-                matricesData[index + 0] = this._positions[i * 3 + 0];
-                matricesData[index + 1] = this._positions[i * 3 + 1];
-                matricesData[index + 2] = this._positions[i * 3 + 2];
-
-                matricesData[index + 4] = this._uBuffer[32 * i + 24 + 0] / 255;
-                matricesData[index + 5] = this._uBuffer[32 * i + 24 + 1] / 255;
-                matricesData[index + 6] = this._uBuffer[32 * i + 24 + 2] / 255;
-                matricesData[index + 7] = this._uBuffer[32 * i + 24 + 3] / 255;
-
-                matricesData[index + 8] = this._covA[i * 3 + 0];
-                matricesData[index + 9] = this._covA[i * 3 + 1];
-                matricesData[index + 10] = this._covA[i * 3 + 2];
-
-                matricesData[index + 12] = this._covB[i * 3 + 0];
-                matricesData[index + 13] = this._covB[i * 3 + 1];
-                matricesData[index + 14] = this._covB[i * 3 + 2];
+                splatIndex[j] = indexMix[2 * j];
             }
-
-            this.mesh?.thinInstanceBufferUpdated("matrix");
+            this.mesh?.thinInstanceBufferUpdated("splatIndex"); // update splatIndex only
         };
 
         // update so this.mesh is valid when exiting this function
         this.mesh = this._getMesh(this.scene);
-        this.mesh.thinInstanceSetBuffer("matrix", matricesData, 16, false);
+        this.mesh.thinInstanceSetBuffer("matrix", matricesData, 1, true);
+        this.mesh.thinInstanceSetBuffer("splatIndex", splatIndex, 1, false);
+
+        /// create textures for gaussian info
+        if (this._material.name == "GaussianSplattingShader") {
+            let material = this.mesh.material as ShaderMaterial;
+
+            let textureSize = this.getTextureSize(this.vertexCount);
+            material.setVector2("dataTextureSize", textureSize);
+
+            let convATexture = new RawTexture(this._covA, textureSize.x, textureSize.y, Engine.TEXTUREFORMAT_RGB, this.scene, false, false, Engine.TEXTURE_BILINEAR_SAMPLINGMODE, Engine.TEXTURETYPE_FLOAT); // floating issue

```

Comment:
Please use `Constants` instead of `Engine` when referencing static flags, as they are later being replaced with the value instead of from the object itself. i.e.:

```suggestion
            let convATexture = new RawTexture(this._covA, textureSize.x, textureSize.y, Constants.TEXTUREFORMAT_RGB, this.scene, false, false, Constants.TEXTURE_BILINEAR_SAMPLINGMODE, Constants.TEXTURETYPE_FLOAT); // floating issue
```

There are other references to the Engine class at later points in the code as well